src/reusedBrowser.ts

```diff
@@ -111,28 +111,52 @@ export class ReusedBrowser implements vscodeTypes.Disposable {
     if (!backend)
       return { errors };
     backend.onClose(() => {
-      if (backend === this._backend) {
-        this._backend = undefined;
+      if (backend === this._testingBackend) {
+        this._testingBackend = undefined;
         void this._reset('none');
```

Comment:
I think this logic should later apply to the "currently recording backend", once you implement recording in all backends.